pkg/service/model_build_target_group.go

```diff
@@ -0,0 +1,392 @@
+	targetType elbv2model.TargetType, tgProtocol elbv2model.Protocol, hc *elbv2model.TargetGroupHealthCheckConfig) string {
+	healthCheckProtocol := string(elbv2model.ProtocolTCP)
+	healthCheckInterval := strconv.FormatInt(t.defaultHealthCheckInterval, 10)
+	if hc.Protocol != nil {
+		healthCheckProtocol = string(*hc.Protocol)
+	}
+	if hc.IntervalSeconds != nil {
+		healthCheckInterval = strconv.FormatInt(*hc.IntervalSeconds, 10)
+	}
+	uuidHash := sha256.New()
+	_, _ = uuidHash.Write([]byte(t.clusterName))
+	_, _ = uuidHash.Write([]byte(t.service.UID))
+	_, _ = uuidHash.Write([]byte(strconv.Itoa(int(tgPort))))
+	_, _ = uuidHash.Write([]byte(svcPort.String()))
+	_, _ = uuidHash.Write([]byte(targetType))
+	_, _ = uuidHash.Write([]byte(tgProtocol))
+	_, _ = uuidHash.Write([]byte(healthCheckProtocol))
+	_, _ = uuidHash.Write([]byte(healthCheckInterval))
+	uuid := hex.EncodeToString(uuidHash.Sum(nil))
+
+	sanitizedNamespace := invalidTargetGroupNamePattern.ReplaceAllString(t.service.Namespace, "")
+	sanitizedName := invalidTargetGroupNamePattern.ReplaceAllString(t.service.Name, "")
+	return fmt.Sprintf("k8s-%.8s-%.8s-%.10s", sanitizedNamespace, sanitizedName, uuid)
+}
+
+func (t *defaultModelBuildTask) buildTargetGroupAttributes(_ context.Context) ([]elbv2model.TargetGroupAttribute, error) {
+	var rawAttributes map[string]string
+	if _, err := t.annotationParser.ParseStringMapAnnotation(annotations.SvcLBSuffixTargetGroupAttributes, &rawAttributes, t.service.Annotations); err != nil {
+		return nil, err
+	}
+	if rawAttributes == nil {
+		rawAttributes = make(map[string]string)
+	}
+	if _, ok := rawAttributes[tgAttrsProxyProtocolV2Enabled]; !ok {
+		rawAttributes[tgAttrsProxyProtocolV2Enabled] = strconv.FormatBool(t.defaultProxyProtocolV2Enabled)
+	}
+	proxyV2Annotation := ""
+	exists := t.annotationParser.ParseStringAnnotation(annotations.SvcLBSuffixProxyProtocol, &proxyV2Annotation, t.service.Annotations)
+	if exists {
+		if proxyV2Annotation != "*" {
+			return []elbv2model.TargetGroupAttribute{}, errors.Errorf("invalid value %v for Load Balancer proxy protocol v2 annotation, only value currently supported is *", proxyV2Annotation)
+		}
+		rawAttributes[tgAttrsProxyProtocolV2Enabled] = "true"
+	}
+	attributes := make([]elbv2model.TargetGroupAttribute, 0, len(rawAttributes))
+	for attrKey, attrValue := range rawAttributes {
+		attributes = append(attributes, elbv2model.TargetGroupAttribute{
+			Key:   attrKey,
+			Value: attrValue,
+		})
+	}
+	return attributes, nil
+}
+
+func (t *defaultModelBuildTask) buildPreserveClientIPFlag(_ context.Context, targetType elbv2model.TargetType) (bool, error) {
+	var rawAttributes map[string]string
+	if _, err := t.annotationParser.ParseStringMapAnnotation(annotations.SvcLBSuffixTargetGroupAttributes, &rawAttributes, t.service.Annotations); err != nil {
+		return false, err
+	}
+	if rawVal, ok := rawAttributes[tgAttrsPreserveClientIPEnabled]; ok {
```

Comment:
i think it would be better to move this validation logic of bool type into buildTargetGroupAttributes. since it validate a targetGroup attribute.
and change this method to accept targetType and []elbv2model.TargetGroupAttribute

Reply:
I will modify this function to take []elbv2model.TargetGroupAttribute, and move the validation logic to buildTargetGroupAttributes. Since elbv2model.TargetGroupAttribute contains string key, value pairs, the strconv.ParseBool, and error return is still needed.